File: src/main/java/com/foodorderapp/models/entity/Product.java
package com.foodorderapp.models.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "products")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Product extends BaseEntity {

    @Column(name = "name", unique = true, nullable = false)
    private String name;
    @Column(name = "description", nullable = false)
    private String content;
    @Column(name = "type", nullable = false)
    private String type;
    @Column(name = "volume", nullable = false)
    private Integer volume;
    @Column(name = "price", nullable = false)
    private double price;
    @Column(name = "available")
    private Boolean available;
    @Lob
    private byte[] image;
}

File: src/main/ui/food-order-app/src/app/models/product.ts
import { ProductType } from "../enums/product-type";
import { IProduct } from "../interfaces/product";

export class Product implements IProduct {
    id: string;
    name: string;
    content: string;
    volume: number;
    price: number;
    type: ProductType;
    picture: string;
    constructor() { }
}
